labgrid/driver/shelldriver.py

```diff
@@ -110,6 +110,35 @@ def _run(self, cmd, *, timeout=30.0, codec="utf-8", decodeerrors="strict"):
     def run(self, cmd, timeout=30.0, codec="utf-8", decodeerrors="strict"):
         return self._run(cmd, timeout=timeout, codec=codec, decodeerrors=decodeerrors)
 
+    def _raw_run(self, cmd, timeout=2, codec="utf-8", decodeerrors="strict"):
+        """Runs command without run() wrapper as it's not yet available"""
+        marker = gen_marker()
+        self.console.sendline(f"echo {marker[:4]}; {cmd} ; echo {marker[4:]}")
+        _, _, match, _ = self.console.expect(
+            rf"{marker[:4]}(.*){marker[4:]}\s+{self.prompt}",
+            timeout=timeout
+        )
+        data = self.re_vt100.sub('', match.group(1).decode(codec, decodeerrors)).split('\r\n')
+        if data and not data[-1]:
+            del data[-1]
+            return data
+
+    def _silence_kernel(self, timeout, kernel_log_level=1, codec="utf-8", decodeerrors="strict"):
+        """Tries to set kernel log level to KERN_ALERT log level to make it less chatty."""
+
+        while True:
+            self.console.sendline(f"dmesg -n {kernel_log_level}")
+            try:
+                data = self._raw_run("cat /proc/sys/kernel/printk")
+                if data[2] and data[2].startswith(f'{kernel_log_level}\t'):
+                    return
+
+            except TIMEOUT:
+                pass
+
+            if timeout.expired:
+                raise TIMEOUT(f"Timeout of {self.login_timeout} seconds exceeded during waiting for kernel silencer")  # pylint: disable=line-too-long
+
```

Comment:
I'm not sure we want more retry code for certain devices behaving strangely. Shouldn't these devices wait until input can be handled reliably before starting login/shell?

From the commit message:
> Those issues mostly happen frequently during early boot process, where there is a lot of output
going on.

Could `await_login_timeout` help here instead?